cpp/modmesh/universe/polygon.hpp

```diff
@@ -969,6 +973,26 @@ class PolygonPad
     using curve_pad_type = CurvePad<T>;
     using rtree_type = RTree<segment_type, BoundBox3d<T>, RTreeValueOps<segment_type, BoundBox3d<T>>>;
 
+    /**
+     * Trapezoid decomposition structure for boolean operations.
+     *
+     * A trapezoid is a quadrilateral with two parallel horizontal edges (top and bottom).
+     * In polygon decomposition, vertical lines are extended from each vertex to partition
+     * the polygon into trapezoids. This representation simplifies polygon boolean operations
+     * by reducing complex polygons to a set of simple trapezoids that can be easily
+     * classified and merged.
+     *
+     * Uses sweep line algorithm to partition polygon into trapezoids.
+     */
+    struct Trapezoid
+    {
+        value_type top_y; ///< Y-coordinate of the top horizontal edge
+        value_type bottom_y; ///< Y-coordinate of the bottom horizontal edge
+        segment_type left_edge; ///< Left boundary segment of the trapezoid
+        segment_type right_edge; ///< Right boundary segment of the trapezoid
+        size_t source_polygon; ///< ID of the polygon this trapezoid came from
+    };
```

Comment:
Struct for trapazoid

Reply:
`Trapezoid` should be taken outside to be a normal class (with the name `Trapezoid3d`, `class` declaration, private data with prefix, and accessors).  It will be so useful that you may just create `TrapezoidPad` using this PR.